pkg/clustermesh/common/remote_cluster.go

```diff
@@ -57,6 +57,9 @@ type remoteCluster struct {
 	// connect to the etcd cluster of the remote cluster
 	configPath string
 
+	// mesh is the cluster mesh this remote cluster belongs to
+	mesh *ClusterMesh
+
```

Comment:
I'd personally avoid introducing the circular dependency on ClusterMesh here, unless strictly necessary. I know that we already have it in the `pkg/clustermesh/remote_cluster` structure, but it looks like a code smell and it makes more difficult to figure out which fields are actually accessed.

Reply:
I didn't feel great about this either, but we won't need this after I move the validation to clustermesh/remoteCluster.Run :+1: